### PR TITLE
[10.6.X][GEOMETRY] Fix deps to resolve undefined reference mentioned in UBSAN IBS

### PR DESCRIPTION
**Pull request by smuzaffar**
To resolve
```
tmp/slc7_amd64_gcc700/src/Geometry/GlobalTrackingGeometryBuilder/test/testGlobalTrackingGeometry/GlobalTrackingGeometryTest.cc.o:(.data.rel+0x10d8): undefined reference to `typeinfo for MTDGeometry'
```

### CODE REVIEW

**Comment by smuzaffar** [comment]
please test

File: Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml [hunk 1]
```diff
@@ -5,6 +5,7 @@
 <use   name="Geometry/RPCGeometry"/>
 <use   name="Geometry/DTGeometry"/>
 <use   name="Geometry/CSCGeometry"/>
+<use   name="Geometry/MTDGeometryBuilder"/>
```
Comment (ianna) [inline Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml]: @smuzaffar - IMHO, this is wrong. There should not be any dependencies on the geometry builders.

File: DetectorDescription/DDCMS/BuildFile.xml [hunk 2]
```diff
@@ -3,6 +3,7 @@
 <use   name="FWCore/Utilities"/>
 <use   name="dd4hep"/>
 <use   name="rootmath"/>
+<use   name="rooteve"/>
```
Comment (ianna) [inline DetectorDescription/DDCMS/BuildFile.xml]: @smuzaffar - I'm not sure we want to depend on Eve. I can check it next year.

**Comment by ianna** [comment]
@smuzaffar - If it's not crucial for a release, please, leave it unfixed. It'd be better to open an issue. Thanks.

**Review by ianna — CHANGES_REQUESTED** [review]

File: Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml (hunk 1 above)
Comment (Dr15Jones) [reply Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml]: I agree with Yana.

File: DetectorDescription/DDCMS/BuildFile.xml (hunk 2 above)
Comment (Dr15Jones) [reply DetectorDescription/DDCMS/BuildFile.xml]: I agree with Yana.

Comment (Dr15Jones) [reply DetectorDescription/DDCMS/BuildFile.xml]: I think we need a new root package which has as its top `libGeo`.

File: Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml (hunk 1 above)
Comment (fabiocos) [reply Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml]: @ianna @Dr15Jones so what is the preferred solution here? Integrate the code as it is and then modify the Tracker/MTD structure in next future, or keep this on hold until the geometries have been modified?

**Comment by fabiocos** [comment]
please abort

File: Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml (hunk 1 above)
Comment (ianna) [reply Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml]: I'd suggest to restructure before integrating the PR

File: DetectorDescription/DDCMS/BuildFile.xml (hunk 2 above)
Comment (smuzaffar) [reply DetectorDescription/DDCMS/BuildFile.xml]: PR https://github.com/cms-sw/cmssw/pull/25566 should resolve it.

File: Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml (hunk 1 above)
Comment (smuzaffar) [reply Geometry/GlobalTrackingGeometryBuilder/test/BuildFile.xml]: @ianna , this dependency is in tests only. Looks like this unit test is testing many other geometries too.

[a]
```
  void analyzeCSC(const GlobalTrackingGeometry* geo, const CSCGeometry* cscGeometry);
  void analyzeDT(const GlobalTrackingGeometry* geo, const DTGeometry* dtGeometry);
  void analyzeRPC(const GlobalTrackingGeometry* geo, const RPCGeometry* rpcGeometry);
  void analyzeGEM(const GlobalTrackingGeometry* geo, const GEMGeometry* gemGeometry);
  void analyzeMTD(const GlobalTrackingGeometry* geo, const MTDGeometry* mtdGeometry);
  void analyzeTracker(const GlobalTrackingGeometry* geo, const TrackerGeometry* tkGeometry);
```

**Comment by smuzaffar** [comment]
@ianna , I have updated the PR. Now there is separate PR #25856 for fixing the libGeom dependency. And for this PR, the dependency is in unit tests which is testing many other geometries too ( see  https://github.com/cms-sw/cmssw/pull/25566#discussion_r253547573 )

**Comment by smuzaffar** [comment]
please test
@ianna ??

**Comment by ianna** [comment]
@fabiocos - if the following class is moved to the plugins directory there should not be any problem with linking against the library.
```
https://cmssdt.cern.ch/lxr/source/Geometry/MTDGeometryBuilder/interface/MTDGeomBuilderFromGeometricTimingDet.h
https://cmssdt.cern.ch/lxr/source/Geometry/MTDGeometryBuilder/src/MTDGeomBuilderFromGeometricTimingDet.cc
```
@cvuosalo - FYI

**Comment by ianna** [comment]
+1

Move of the files can be done later.

**Comment by smuzaffar** [comment]
@fabiocos , can we get this in or should I close the PR?

**Comment by fabiocos** [comment]
@smuzaffar I have tracked the problem into an issue to keep memory of this, for the time being I would say we may close this PR

**Pull request closed without merging**